dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -1813,7 +1814,7 @@ protected List<TaskInstance> getRecoverTaskInstanceList(String cmdParam) {
         if (paramMap != null && paramMap.containsKey(CMD_PARAM_RECOVERY_START_NODE_STRING)) {
             String[] idList = paramMap.get(CMD_PARAM_RECOVERY_START_NODE_STRING).split(Constants.COMMA);
             if (ArrayUtils.isNotEmpty(idList)) {
-                List<Integer> taskInstanceIds = Arrays.stream(idList).map(Integer::valueOf).collect(Collectors.toList());
+                List<Integer> taskInstanceIds = Arrays.stream(idList).filter(StringUtils::isNotEmpty).map(Integer::valueOf).collect(Collectors.toList());
```

Comment:
It's better to check empty id when setting, so we don't have to filter it anywhere we use.

Reply:
It's nice, ok.